controllers/apps/operations/restart.go

```diff
@@ -113,32 +119,92 @@ func (r restartOpsHandler) podApplyCompOps(
 	return !pod.CreationTimestamp.Before(&ops.Status.StartTimestamp)
 }
 
-// restartStatefulSet restarts statefulSet workload
-func (r restartOpsHandler) restartComponent(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource, objList client.ObjectList) error {
-	if err := cli.List(reqCtx.Ctx, objList,
-		client.InNamespace(opsRes.Cluster.Namespace),
-		client.MatchingLabels{constant.AppInstanceLabelKey: opsRes.Cluster.Name}); err != nil {
-		return err
+func (r restartOpsHandler) getComponentOrders(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) ([]appsv1alpha1.ComponentOps, error) {
+	cd := &appsv1alpha1.ClusterDefinition{}
+	if opsRes.Cluster.Spec.ClusterDefRef == "" || opsRes.Cluster.Spec.Topology == "" {
+		return nil, nil
+	}
+	if err := cli.Get(reqCtx.Ctx, client.ObjectKey{Name: opsRes.Cluster.Spec.ClusterDefRef}, cd); err != nil {
+		return nil, err
 	}
+	// components that require sequential restart
+	var orderedComps []appsv1alpha1.ComponentOps
+	for _, topology := range cd.Spec.Topologies {
+		if topology.Name != opsRes.Cluster.Spec.Topology {
+			continue
+		}
+		if topology.Orders != nil && len(topology.Orders.Update) > 0 {
+			// when using clusterDef and topology, "update orders" includes all components
+			for _, compName := range topology.Orders.Update {
```

Comment:
should we add a restart order ? @leon-inf

Reply:
Is there any special requirement to define a restart order specifically for it?